### ISSUE TITLE
Content-Type Requires Splitting before Check

### ISSUE THREAD

**Issue by cpoliver** [ferdikoomen/openapi-typescript-codegen#386]
The generated `getResponseBody()` function does not account for OData metadata values set in the Content-Type header.

This is defined in the [OData standard](http://docs.oasis-open.org/odata/odata-json-format/v4.0/os/odata-json-format-v4.0-os.html#_Toc372793046) which is an ISO/IEC International Standard.

The example where I'm encountering this has the content-type followed by some additional information which should be ignored when determining the content-type.

The code currently does this:

```typescript
const contentType = response.headers.get('Content-Type'); // 'application/json; odata.metadata=minimal; odata.streaming=true'

if (contentType) {
	switch (contentType.toLowerCase()) {
		case 'application/json':
    	case 'application/json; charset=utf-8':
    		return await response.json();
		default:
			return await response.text();
	}
}
}
```

Whereas it might be better off doing something like this:

```typescript
	const contentType = response.headers.get('Content-Type'); // 'application/json; odata.metadata=minimal; odata.streaming=true'

	const mediaType = contentType?.split(';')[0].toLowerCase() // 'application/json'

	if (mediaType === 'application/json') {
		return await response.json();
	}

	return await response.text();
}
```

Example in editor:

![image](https://user-images.githubusercontent.com/1554333/96085980-7f280580-0eb9-11eb-9b4a-241497d485ec.png)

**Comment by ferdikoomen** [comment]
Oh my god, what are they doing with that poor header :-)
I'll have a quick look, you spit method makes sense, i was thinking we could also use `contentType.includes('application/json')` but maybe that is a bit tricky...

**Comment by cpoliver** [comment]
Haha! You're right, I might have overengineered it, a simple `includes` or `startsWith` should suffice. Thanks again 👍

**Comment by ferdikoomen** [comment]
@cpoliver i added a fix for this, i will push a 0.5.3 version in a bit, but want to fix a few small things as well and test the release.

**Comment by ferdikoomen** [comment]
Pushed 0.5.3 to NPM
